src/services/git/GitServiceUtils.ts

```diff
@@ -1,16 +1,18 @@
 import gitUrlParse from 'git-url-parse';
-import { GitService } from './model';
+import _ from 'lodash';
+import { ScanningStrategy } from '../../detectors';
+import { ServiceType } from '../../detectors/IScanningStrategy';
 import { assertNever } from '../../lib/assertNever';
+import { ProjectComponent } from '../../model';
 
 export class GitServiceUtils {
-  static getUrlToRepo = (url: string, path?: string | undefined, branch = 'master') => {
+  static getUrlToRepo = (url: string, scanningStrategy: ScanningStrategy, path?: string | undefined, branch = 'master') => {
     const parsedUrl = gitUrlParse(url);
 
     let completeUrl = `${parsedUrl.protocol}://${parsedUrl.resource}/${parsedUrl.owner}/${parsedUrl.name}`;
-    const sourceUrl = <GitService | null>parsedUrl.resource;
 
-    if (path && sourceUrl) {
-      completeUrl += GitServiceUtils.getPath(sourceUrl, path, branch || parsedUrl.ref);
+    if (path) {
+      completeUrl += GitServiceUtils.getPath(path, branch || parsedUrl.ref, <ServiceType>scanningStrategy.serviceType);
```

Comment:
is it necessary to re-type?

Reply:
```suggestion
      completeUrl += GitServiceUtils.getPath(path, branch || parsedUrl.ref, scanningStrategy.serviceType);
```

Reply:
ServiceType can be undefined at first, so that's the reason why I retype it. We have the ServiceType at this time.

@prokopsimek

Reply:
@adelkahomolova I guess it's better to use `scanningStrategy.serviceType!` instead of re-typing. The type-checking control won't work if something wrong happens with the ServiceType interface.